libsolidity/formal/PredicateSort.h

```diff
@@ -37,7 +37,7 @@ namespace solidity::frontend::smt
  *
  * 2. Nondet interface
  * The nondeterminism behavior of a contract. Signature:
- * nondet_interface(blockchainState, stateVariables, blockchainState', stateVariables').
+ * nondet_interface(blockchainState, stateVariables, blockchainState', stateVariables', error, this).
```

Comment:
Why do we need `this` but not `txData`, `crypto` etc?

Reply:
We probably *do* need this, but we don't have examples for that yet.
This is the minimal change for the known benchmarks to work correctly.
Also, for the `transaction data` we need to be careful, since the external call creates a new transaction, right?

Reply:
I have added `crypto` and `abi` to the `nondet_interace` and added a test to check `crypto` across external calls.
As discussed, `txData` should not be part of `nondet_interface`, similarly to the `interface`.